src/websocket-client.ts

```diff
@@ -1197,7 +1207,7 @@ export class WebsocketClient extends EventEmitter {
 
   public subscribeEndpoint(
     endpoint: string,
-    market: 'spot' | 'usdm' | 'coinm',
+    market: 'spot' | 'spotTestnet' | 'usdm' | 'coinm',
     forceNewConnection?: boolean,
   ): WebSocket {
     const wsKey = getWsKeyWithContext(market, endpoint);
```

Comment:
So with my proposed change of having the same `useTestnet` boolean in the ws configuration object, any fn like this would basically remain as before:
```typescript
  public subscribeEndpoint(
    endpoint: string,
    market: 'spot' | 'usdm' | 'coinm',
    forceNewConnection?: boolean,
  ): WebSocket {
  ```

The user wouldn't need to say in the function to use spotTestnet instead of spot, they would just say "spot" and the SDK automatically knows to use testnet, because that's in the WS client's config in the constructor. Not a breaking change, since "spotTestnet" was never a market choice in the function call anyway, and this way it's less for the user to think about while working similar to how it works in the REST client with the change you've just added.

The only caveat is each line like this:
```typescript
    const wsKey = getWsKeyWithContext(market, endpoint);
```

might need to become something like this (open to cleaner ideas):

```typescript
    const resolvedMarket = this.options.useTestnet ? market + 'Testnet' : market;
    const wsKey = getWsKeyWithContext(resolvedMarket, endpoint);
```

So this way the user only cares about testnet when instancing the WS client, and the rest of the function calls are the same for live vs testnet. It also makes the new "isTestnet?: boolean," param in some of the functions further down, redundant. Let me know your thoughts! The less a user has to worry about internal plumbing, the less mistakes they make -> less confusion -> easier usage all around :)